### PR TITLE
[Keras] Identity Loss function

### LINKED ISSUE #587: Investigate changes needed for custom loss functions

### PR DESCRIPTION
**Pull request by soumyac1999**
**Description of changes:**

**Related Issues:**

Linked Issues:
- Issue #587

Issues closed by this PR:
- Closes #587 (for now)

**Before merging:**

- [ ] Did you update the [flexflow-third-party](https://github.com/flexflow/flexflow-third-party) repo, if modifying any of the Cmake files, the build configs, or the submodules?

### CODE REVIEW

**Comment by soumyac1999** [comment]
We might want to discuss the API (y is passed to the fit function but not used in the loss computation but only for computing metrics). Please refer to the example at https://github.com/flexflow/FlexFlow/pull/602/files#diff-11505aef1f6b50568c746acf893725321d36e6ef9c6ce07a5c97e30fbff3c459. 

Or we could merge this as is and later add support for arbitrary loss functions like keras does (which shouldn't be too much work with the identity loss in place).

**Comment by lockshaw** [comment]
LGTM, though I'd like @jiazhihao's thoughts as well as I'm less familiar with the loss code. 

Also @gabrieleoliaro looks like there are some issues with prebuild downloading in CI--does this need to be fixed?

**Review by jiazhihao — APPROVED** [review]
